PurchasesCoreSwift/Caching/InMemoryCachedObject.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  InMemoryCachedObject.swift
+//  PurchasesCoreSwift
+//
+//  Created by Joshua Liebowitz on 7/13/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+open class InMemoryCachedObject<T> {
+
+    private let accessQueue = DispatchQueue(label: "InMemoryCachedObjectQueue", attributes: .concurrent)
```

Comment:
Unfortunately, unifying this queue doesn't make as much sense since this is used exclusively for maintaining exclusive access to a datasource. If we used a different queue for this, we might end up deadlocking if a work item in a global queue ends up doing a read/write to the datasource via `.sync`. This queue is a special purpose one, we really need to have very close control over it to avoid that sort of thing. Now that it is serial, and we're only dispatching `sync`, resource usage should be minimal.

Reply:
same question here regarding `concurrent + .barrier` versus `serial`

Reply:
Same answer, `.barrier` is less common and gets people to think about what is going on more. It's a signal. I'm leaving these non-harmful things in to trigger people (probably myself) into thinking about what is happening more.

Reply:
🤔 if the `.barrier` flag isn't doing anything other than signaling intent, perhaps a clearer way to signal the intent would be to create a wrapper with a clear name? 

i.e.: 
`accessQueue.sync(flags: .barrier) {` -> `accessQueue.executeByLockingDatasource {`?

Reply:
I'd still keep a comment to clarify what's going on

Reply:
Omg, great idea! Gonna do this.